scripts/release-checks/dependency-ranges/peer-deps-check.mts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import path from 'path';
+import url from 'url';
+import semver from 'semver';
+import { PackageMap } from '../../../lib/packages.js';
+
+/** Path to the current directory. */
+const currentDir = path.dirname(url.fileURLToPath(import.meta.url));
+
+/** Path to the project directory. */
+const projectDir = path.join(currentDir, '../../../');
+
+/** Describes a parsed `package.json` file. */
+interface PackageJson {
+  name?: string;
+  peerDependencies?: Record<string, string>;
+}
+
+export async function checkPeerDependencies(
+  newVersion: semver.SemVer,
+  allPackages: PackageMap,
+): Promise<string[]> {
+  const { major, minor } = newVersion;
+  const isPrerelease = !!newVersion.prerelease[0];
+  const isMajor = minor === 0;
+
+  let expectedFwPeerDep = `^${major}.0.0`;
+  if (isMajor && isPrerelease) {
+    expectedFwPeerDep = `^${major}.0.0-next.0`;
+  } else if (isPrerelease) {
+    expectedFwPeerDep = `^${major}.0.0 || ^${major}.${minor}.0-next.0`;
+  }
+
+  const failures: string[] = [];
+  for (const pkgInfo of Object.values(allPackages)) {
+    failures.push(...checkPackage(pkgInfo.packageJson, expectedFwPeerDep));
+  }
```

Comment:
Consider:

```typescript
const failures = Object.values(allPackages)
    .flatMap((pkgInfo) => checkPackage(pkgInfo.packageJson, expectedFwPeerDep));
```

Reply:
Fair enough. I think it comes down to personal preference. Please feel free to make this change if you feel like it